### DISCUSSION TITLE
Add "kid" in JWT header parameters when sign access_token

### DISCUSSION THREAD

**Discussion by llitllie** [directus/directus#17840] [Feature Requests]
### Summary

The JWT header of Directus access token only contains "alg" and "typ", which doesn't satisfy JWS.
Add "kid" into JWT header when sign/refresh "access_token". Then the other apps could use the "kid" to matche the  key in JWK file and verify the token. 

It's better to read this variable from the config file.

### Basic Example

The JWT headers should containt "kid" in its header
```
{
    "alg": "RS256",
    "typ": "JWT",
    "kid": "directus"
}
```

### Motivation

We're building an app not only integrate with Directus but others, and we'd like to use Directus access token to access others APIs.  And others API may verify the token locally. Share the JWT secrect key may archive this in some cases, but some apps not - they require hige security . For example, Krakend acts a API gateway and it only use JWS, which requires "kid" in JWT header, see: https://www.krakend.io/docs/authorization/jwt-validation/#jwt-header-requirements.

### Detailed Design

Maybe add "kid" into JWT header when sign/refresh "access_token".
Configure "kid" in .env file
```
JWS_KEYID="direct"
```
Sign access token with "kid"
```ts
		const accessToken = jwt.sign(customClaims, env.SECRET as string, {
			expiresIn: env.ACCESS_TOKEN_TTL,
			issuer: 'directus',
			keyid: env.JWS_KEYID,
		});
```

### Must Have:
- specify "kid" in JWT header 

### Should Have:
- configure "kid" in environment variables 

### Drawbacks

The code changes looks like small but may add complexit to code base.

### Alternatives

Directus may provide a hook to filter custom sign options. Other solutions like buidl a new auth route may take more efforts.

### Adoption Strategy

It's proposal to add a new key in JWT header and the auth will continue work without impact any existing Directus features/projects. Also, tt's not a breaking change.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
